pkg/ui/v1alpha2/backend.go

```diff
@@ -0,0 +1,493 @@
+package ui
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	experimentv1alpha2 "github.com/kubeflow/katib/pkg/api/operators/apis/experiment/v1alpha2"
+	api_pb_v1alpha2 "github.com/kubeflow/katib/pkg/api/v1alpha2"
+	common_v1alpha2 "github.com/kubeflow/katib/pkg/common/v1alpha2"
+
+	"github.com/kubeflow/katib/pkg/util/v1alpha2/katibclient"
+	"google.golang.org/grpc"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/ghodss/yaml"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func NewKatibUIHandler() *KatibUIHandler {
+	kclient, err := katibclient.NewClient(client.Options{})
+	if err != nil {
+		log.Printf("NewClient for Katib failed: %v", err)
+		panic(err)
+	}
+	return &KatibUIHandler{
+		katibClient: kclient,
+	}
+}
+
+func (k *KatibUIHandler) connectManager() (*grpc.ClientConn, api_pb_v1alpha2.ManagerClient) {
+	conn, err := grpc.Dial(common_v1alpha2.ManagerAddr, grpc.WithInsecure())
+	if err != nil {
+		log.Printf("Dial to GRPC failed: %v", err)
+		return nil, nil
+	}
+	c := api_pb_v1alpha2.NewManagerClient(conn)
+	return conn, c
+}
+
+func (k *KatibUIHandler) FetchHPJobs(w http.ResponseWriter, r *http.Request) {
+	//enableCors(&w)
+
+	jobs := make([]JobView, 0)
+
+	el, err := k.katibClient.GetExperimentList()
```

Comment:
I perfer we add `filter` in `GetExperimentList` db interface

Reply:
Do you mean `filter` for HP or NAS jobs?

Reply:
yes, filter can help us filter we need in db server instead of our code implementation

Reply:
Here we use Katib client, not db interface function. We can add filter in `GetExperimentList` in Katib Client, but is it make sense?